tests/module_cusolver/CMakeLists.txt

```diff
@@ -0,0 +1,110 @@
+########################################
+# CMake build system
+# This file is part of ABACUS
+cmake_minimum_required(VERSION 3.18)
+########################################
+# project name
+project(module_cusolver_test)
```

Comment:
Is it possible modifying `AddTest` function of root CMakeLists file (adding CUDA libs, etc.) to build `module_cusolver_test` without another project?

Reply:
Do you mean the executable files should be generated at build/source/src_pdiag/test/ when BUILD_TESTING is on? By the way, would you explain the tests directory and I want to know what should be added in this directory.

Reply:
> Do you mean the executable files should be generated at build/source/src_pdiag/test/ when BUILD_TESTING is on? 

Yes, I suggest we following the global UT rules.

> By the way, would you explain the tests directory and I want to know what should be added in this directory.

The test directory should be contained in its corresponding module directory. It includes all test codes for this module(also `CMakeLists.txt` to build them). You can read [docs](https://github.com/deepmodeling/abacus-develop/blob/develop/doc/developers.md#add-a-unit-test) for more details.

Reply:
> > By the way, would you explain the tests directory and I want to know what should be added in this directory.
> 
> The test directory should be contained in its corresponding module directory. It includes all test codes for this module(also `CMakeLists.txt` to build them). You can read [docs](https://github.com/deepmodeling/abacus-develop/blob/develop/doc/developers.md#add-a-unit-test) for more details.

Sorry for my ambiguousness and the tests directory mentioned above is `root/tests`.
<img width="180" alt="image" src="https://user-images.githubusercontent.com/25193323/159415516-9f0d94a8-6d7d-4ae6-aa34-481cecbf176d.png">
<img width="545" alt="image" src="https://user-images.githubusercontent.com/25193323/159415553-2d78ee8d-32d4-4a21-a19f-a896bd56a7e2.png">

Reply:
Because I only find `CMakelists.txt` in `module_base` without `src` while  `src` in `module_cell` without `CMakelists.txt`.

Reply:
> Sorry for my ambiguousness and the tests directory mentioned above is `root/tests`.

This dir is only for E2E tests. Other components are not in use.

Reply:
> Because I only find `CMakelists.txt` in `module_base` without `src` while `src` in `module_cell` without `CMakelists.txt`.

Our UT is still building, and `module_base` would be a better example for you than `module_cell`.